packages/blocks/src/blocks/CoreImage.tsx

```diff
@@ -0,0 +1,104 @@
+import { gql } from '@apollo/client';
+import React from 'react';
+import Image from 'next/image.js';
+import { useBlocksTheme } from '../components/WordPressBlocksProvider.js';
+import { ContentBlock } from '../components/WordPressBlocksViewer.js';
+import { getStyles } from '../utils/index.js';
+
+export type CoreImageFragmentProps = ContentBlock & {
+  attributes: {
+    align?: string;
+    alt?: string;
+    anchor?: string;
+    borderColor?: string;
+    caption?: string;
+    className?: string;
+    width?: string;
+    url?: string;
+    title?: string;
+    style?: string;
+    src?: string;
+    sizeSlug?: string;
+    rel?: string;
+    lock?: string;
+    linkTarget?: string;
+    linkDestination?: string;
+    linkClass?: string;
+    id?: string;
+    href?: string;
+    height?: string;
+    cssClassName?: string;
+  };
+};
+
+export function CoreImage(props: CoreImageFragmentProps) {
+  const theme = useBlocksTheme();
+  const style = getStyles(theme, { ...props });
+  const { attributes } = props;
+
+  if (attributes.width && attributes.height) {
+    return (
+      <figure>
+        <Image
+          style={style}
+          src={attributes.src || ''}
+          width={attributes.width}
+          height={attributes.height}
+          alt={attributes.alt}
+          className={attributes?.cssClassName}
+        />
+        {attributes?.caption && <figcaption>{attributes.caption}</figcaption>}
+      </figure>
+    );
+  }
+  return (
+    <figure>
+      <img
+        style={style}
+        src={attributes.src}
+        width={attributes.width}
+        height={attributes.height}
+        alt={attributes.alt}
+        className={attributes?.cssClassName}
+      />
+      {attributes?.caption && <figcaption>{attributes.caption}</figcaption>}
+    </figure>
+  );
+}
+
+CoreImage.fragments = {
+  key: `CoreImageBlockFragment`,
+  entry: gql`
+    fragment CoreImageBlockFragment on CoreImage {
+      attributes {
+        align
+        alt
+        anchor
+        borderColor
+        caption
+        className
+        width
+        url
+        title
+        style
+        src
+        sizeSlug
+        rel
+        lock
+        linkTarget
```

Comment:
Make sure you handle `linkTarget` and `linkClass`, `linkDestination` attributes here. You need to add a link to an image and then check if a link exists.


Then use `Next/link` component and wrap the image with a link. Check out the following screenshots:

<img width="1231" alt="Screenshot 2023-06-20 at 11 25 51" src="https://github.com/wpengine/faustjs/assets/328805/4f7ffd13-1890-4886-95f7-af6e82655fa4">
<img width="1356" alt="Screenshot 2023-06-20 at 11 23 32" src="https://github.com/wpengine/faustjs/assets/328805/7ab10da4-412d-479b-995d-f49936210996">

<img width="1318" alt="Screenshot 2023-06-20 at 11 26 19" src="https://github.com/wpengine/faustjs/assets/328805/51d50ccd-5cea-4f52-90af-bdb3a0d18c84">

Reply:
Check you how I did this with `Button Block`:

https://github.com/wpengine/faustjs/pull/1461/files#diff-93320378f80ef510ff557c84edb38ec380663c0cefe7ba66b0df3d3a339aaaa7R31-R46

Reply:
@theodesp, thank you for these visuals and the example from CoreButton, I appreciate it. If I add this one to my current conditional in the way you have, I will have four if-conditionals (referencing this https://github.com/wpengine/faustjs/pull/1461/files#:~:text=%3C-,Link,-href%3D%7B )

Is that what you're suggesting here? 

Thanks again for your guidance!

Reply:
@theodesp Image links should be handled. Can you please give this another look?

Reply:
Looks good!